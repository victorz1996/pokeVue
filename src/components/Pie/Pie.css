.centerClass {
    display: flex;
    justify-content: center;
  }
.navbar-text {
    color: black !important;
}
.botonPie {
    border-radius: 10rem;
    color: aliceblue;
    height: 40px;
    width: 330px;
    margin-right: 10px;
    margin-left: 10px;
}
.iconPosition {
    margin-right: 5px;
}