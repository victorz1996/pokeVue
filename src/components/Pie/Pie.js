/**
 * @namespace Pie
 * @description Componente que contiene la informacion que se muestra en el footer
 * de las seccion de listar pokemones
 * @author       Victor Zabala <victorma164@gmail.com>
 */
export default {
    name: "Pie",
    mounted() {
        this.cambioLista('todos')
    },
    methods: {
        /**
        * @memberOf Pie
        * @function cambioLista cambio de lista de pokemones
        * @description Metodo que se ejecuta para cambiar el listado de pokemones que se
        * visualizan
        * @param {Array} lista
        */
        cambioLista(lista) {
            const todos = document.getElementById('todos')
            const favoritos = document.getElementById('favoritos')
            if (lista === 'todos') {
                todos.style.background = '#f22539'
                favoritos.style.background = 'rgb(104, 103, 103)'
                this.$emit('listar', lista)
            }
            if (lista === 'favoritos') {
                todos.style.background = 'rgb(104, 103, 103)'
                favoritos.style.background = '#f22539'
                this.$emit('listar', lista)
            }

        },
    },
}