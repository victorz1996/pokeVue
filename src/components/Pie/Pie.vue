<template>
  <nav class="centerClass navbar fixed-bottom navbar-dark bg-light">
    <button @click="cambioLista('todos')" id="todos" class="btn botonPie">
      <font-awesome-icon class="iconPosition" icon="bars" />All
    </button>
    <button
      @click="cambioLista('favoritos')"
      id="favoritos"
      class="btn botonPie"
    >
      <font-awesome-icon class="iconPosition" icon="star" />Favorites
    </button>
  </nav>
</template>

<script src="./Pie.js"></script>
<style lang="css" src="./Pie.css"></style>