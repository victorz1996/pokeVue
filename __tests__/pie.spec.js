import { shallowMount } from '@vue/test-utils'
import Pie from "../src/components/Pie/Pie.vue"

let cmp

    // eslint-disable-next-line no-undef
    beforeEach(() => {
      cmp = shallowMount(Pie, {
        attachTo: document.body,
      })
    })
    // eslint-disable-next-line no-undef
    describe('Componente Pie', () => {
      // eslint-disable-next-line no-undef
      test('Instancia de Vue.js',() => {
        // eslint-disable-next-line no-undef
        expect(cmp.isVueInstance).toBeTruthy()
      })

      // eslint-disable-next-line no-undef
      it('HTMl renderizado structure esperada',()=>{
        // eslint-disable-next-line no-undef
        expect(cmp.element).toMatchSnapshot()
    })
})