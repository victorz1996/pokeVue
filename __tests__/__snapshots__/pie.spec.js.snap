// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Pie HTMl renderizado structure esperada 1`] = `
<nav
  class="centerClass navbar fixed-bottom navbar-dark bg-light"
>
  <button
    class="btn botonPie"
    id="todos"
  >
    <font-awesome-icon
      class="iconPosition"
      icon="bars"
    />
    All
  
  </button>
   
  <button
    class="btn botonPie"
    id="favoritos"
  >
    <font-awesome-icon
      class="iconPosition"
      icon="star"
    />
    Favorites
  
  </button>
</nav>
`;
